packages/loader/container-loader/src/containerStorageAdapter.ts

```diff
@@ -29,7 +29,12 @@ export class ContainerStorageAdapter implements IDocumentStorageService {
     }
 
     public get policies(): IDocumentStorageServicePolicies | undefined {
-        return this.storageGetter().policies;
+        // back-compat 0.40 containerRuntime requests policies even in detached container if storage is present
+        // and storage is always present in >=0.41.
+        try {
+            return this.storageGetter().policies;
```

Comment:
Are you trying to do `this.storageGetter()?.policies`?
If yes, it's much better to be direct, then catch all exceptions.

Reply:
StorageGetter throws exception if called in detached container. So we can not do "?". We need to catch that here and return undefined.

Reply:
This is just for compat and we will remove this in couple of iterations.